go-runtime/compile/schema.go

```diff
@@ -42,10 +42,11 @@ var (
 )
 
 // NativeNames is a map of top-level declarations to their native Go names.
-type NativeNames map[schema.Decl]string
+type NativeNames map[schema.Node]string
```

Comment:
`schema.Symbol` might be slightly narrower here, if it's usable?

Reply:
hm at the moment `schema.EnumVariant` isn't a `Symbol`. we could change that, but according to this description
```
// A Symbol is a named type that can be referenced by other types. This includes
// user defined data types such as data structures and enums, and builtin types.
```
i'm not sure it qualifies? the underlying type is reference-able from other types, but not really the variant itself

Reply:
I think your analysis is correct. In normal PLs you never reference a variant directly as a type that I can recall, only through the enum to type and match/switch